### PR TITLE
feat(sandbox): add checkbox for translatable property

### PR DESCRIPTION
**Pull request by BibiSebi**
## What?
Added new checkbox 'Translatable' to easily develop by being able to turn on and off the functionality. This PR includes the following changes:

- Adding a new checkbox to manipulate the `translatable` property
- Renaming the `Options` title of a section to `Settings`
- Reflecting changes of `translatable` and `storyLang` inside the code block

Screenshot 2024-02-20 at 16 52 31

## Why?
[JIRA: EXT-2240](https://storyblok.atlassian.net/browse/EXT-2240)

### CODE REVIEW

**Comment by BibiSebi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/361?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#361** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#360** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/361?utm_source=stack-comment).

File: packages/container/src/components/FieldPluginContainer.tsx [hunk 1]
```diff
@@ -111,15 +112,17 @@ const useSandbox = (
   const [schema, setSchema] = useState<FieldPluginSchema>({
     field_type: 'preview',
     options: manifest.options,
+    translatable: false,
   })
   const [content, setContent] = useState<unknown>(initialContent)
-  const [language, setLanguage] = useState<string>('')
+  const [language, setLanguage] = useState<string>('default')
```
Comment (BibiSebi) [inline packages/container/src/components/FieldPluginContainer.tsx:118]: inside storyfront I noticed that the default  language seems to be set to  'default'

File: packages/container/src/components/FieldPluginContainer.tsx [hunk 2]
```diff
@@ -111,15 +112,17 @@ const useSandbox = (
...
   const [stateChangedCallbackId, setStateChangedCallbackId] = useState<string>()
 
   const stateChangedData = useMemo<StateChangedMessage>(
     () => ({
       model: content,
       schema: schema,
+      interfaceLanguage: 'en',
```
Comment (BibiSebi) [inline packages/container/src/components/FieldPluginContainer.tsx:125]: This was added due to type issues of `StateChangedMessage` which forced me to add this line of code.

**Comment by BibiSebi** [comment]
I just realized that this PR should not have been stacked. We will need to release a new version of field-plugin library and then redeploy the sandbox.

**Review by eunjae-lee — APPROVED** [review]
Looks good to me! If you want to merge this later, then I think you could merge #360 first, and run `gt restack` on this branch to rebase on the updated main branch. What do you think?

**Comment by eunjae-lee** [comment]
@BibiSebi anything else you need me to review about this PR? :)

**Comment by BibiSebi** [comment]
@eunjae-lee no, this is just blocked by the field plugin release which is blocked by the storyfront release :D

**Pull request merged**